### PR TITLE
[REACTOS] Always check HalGetBusDataByOffset() return code

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

HalSetBusDataByOffset() too.

JIRA issue: [CORE-12422](https://jira.reactos.org/browse/CORE-12422)

Note:
I noticed this related part on CORE 15545.
This does not implement the EISA case.

## Proposed changes

- Add debug log.
- Return error where it applies.

Note:
I did not touch UniATA.

### CODE REVIEW

**Comment by learn-more** [comment]
Does this fix the linked jira issue?

**Comment by SergeGautherie** [comment]
> Does this fix the linked jira issue?

No: as written, it improves the callers, but does not fix the functions themselves.

**Comment by SergeGautherie** [comment]
Looking for a reviewer...

**Comment by SergeGautherie** [comment]
Looking for an active reviewer...

File: drivers/bus/acpi/osl.c [hunk 1]
```diff
@@ -827,22 +833,29 @@ AcpiOsWritePciConfiguration (
...
-        &buf,
-        Reg,
-        (Width >> 3));
+    Size = HalSetBusDataByOffset(PCIConfiguration,
+                                 PciId->Bus,
+                                 slot.u.AsULONG,
+                                 &buf,
+                                 Reg,
+                                 Width >> 3);
+    if (Size == 0)
```
Comment (tkreuzer) [inline drivers/bus/acpi/osl.c:854]: `if (Size != (width >> 3))`?

File: hal/halx86/mp/ioapic.c [hunk 2]
```diff
@@ -647,13 +648,14 @@ HaliReconfigurePciInterrupts(VOID)
...
-				       &IRQMap[i].DstApicInt,
-				       0x3c /*PCI_INTERRUPT_LINE*/,
-				       1);
-
+         Size = HalSetBusDataByOffset(PCIConfiguration,
+                                      IRQMap[i].SrcBusId,
+                                      (IRQMap[i].SrcBusIrq >> 2) & 0x1f,
+                                      &IRQMap[i].DstApicInt,
+                                      0x3c /* PCI_INTERRUPT_LINE */,
+                                      1);
```
Comment (HBelusca) [inline hal/halx86/mp/ioapic.c:656]:
```suggestion
                                      sizeof(IRQMap[i].DstApicInt));
```
or `sizeof(UCHAR)`

File: hal/halx86/mp/ioapic.c [hunk 3]
```diff
@@ -647,13 +648,14 @@ HaliReconfigurePciInterrupts(VOID)
...
+         if (Size == 0)
```
Comment (HBelusca) [inline hal/halx86/mp/ioapic.c:657]: I wonder whether a comparison with the written size would be better.... `if (Size != sizeof(UCHAR))`

File: drivers/bus/pci/pdo.c [hunk 4]
```diff
@@ -1394,12 +1397,14 @@ PdoStartDevice(
...
-                              &Command,
-                              FIELD_OFFSET(PCI_COMMON_CONFIG, Command),
-                              sizeof(USHORT));
+        Size = HalSetBusDataByOffset(PCIConfiguration,
+                                     DeviceExtension->PciDevice->BusNumber,
+                                     DeviceExtension->PciDevice->SlotNumber.u.AsULONG,
+                                     &Command,
+                                     FIELD_OFFSET(PCI_COMMON_CONFIG, Command),
+                                     sizeof(USHORT));
+        if (Size == 0)
```
Comment (HBelusca) [inline drivers/bus/pci/pdo.c:1406]: I wonder whether a comparison wrt. the expected data size to be written would be better: `if (Size != sizeof(USHORT))`

File: drivers/bus/pci/pdo.c [hunk 5]
```diff
@@ -1394,12 +1397,14 @@ PdoStartDevice(
...
-                              DeviceExtension->PciDevice->SlotNumber.u.AsULONG,
-                              &Command,
-                              FIELD_OFFSET(PCI_COMMON_CONFIG, Command),
-                              sizeof(USHORT));
+        Size = HalSetBusDataByOffset(PCIConfiguration,
+                                     DeviceExtension->PciDevice->BusNumber,
+                                     DeviceExtension->PciDevice->SlotNumber.u.AsULONG,
+                                     &Command,
+                                     FIELD_OFFSET(PCI_COMMON_CONFIG, Command),
+                                     sizeof(USHORT));
```
Comment (HBelusca) [inline drivers/bus/pci/pdo.c:1405]: sizeof(USHORT) or `sizeof(Command)`

File: drivers/bus/pci/pdo.c [hunk 6]
```diff
@@ -1354,12 +1355,14 @@ PdoStartDevice(
...
-                                      DeviceExtension->PciDevice->SlotNumber.u.AsULONG,
-                                      &Irq,
-                                      0x3c /* PCI_INTERRUPT_LINE */,
-                                      sizeof(UCHAR));
+                Size = HalSetBusDataByOffset(PCIConfiguration,
+                                             DeviceExtension->PciDevice->BusNumber,
+                                             DeviceExtension->PciDevice->SlotNumber.u.AsULONG,
+                                             &Irq,
+                                             0x3c /* PCI_INTERRUPT_LINE */,
+                                             sizeof(UCHAR));
```
Comment (HBelusca) [inline drivers/bus/pci/pdo.c:1363]: or `sizeof(Irq)`

File: drivers/bus/pci/pdo.c [hunk 7]
```diff
@@ -1354,12 +1355,14 @@ PdoStartDevice(
...
+                if (Size == 0)
```
Comment (HBelusca) [inline drivers/bus/pci/pdo.c:1364]: and same comment regarding the check as for the others.

File: drivers/bus/acpi/osl.c [hunk 8]
```diff
@@ -808,12 +809,17 @@ AcpiOsReadPciConfiguration (
...
-        Value,
-        Reg,
-        (Width >> 3));
+    Size = HalGetBusDataByOffset(PCIConfiguration,
+                                 PciId->Bus,
+                                 slot.u.AsULONG,
+                                 Value,
+                                 Reg,
+                                 Width >> 3);
+    if (Size == 0)
```
Comment (HBelusca) [inline drivers/bus/acpi/osl.c:818]: See tkreuzer's other comment too.

File: drivers/bus/acpi/osl.c (hunk 1 above)
Comment (SergeGautherie) [reply drivers/bus/acpi/osl.c:854]: Ok, let's try being stricter.

File: hal/halx86/mp/ioapic.c (hunk 2 above)
Comment (SergeGautherie) [reply hal/halx86/mp/ioapic.c:656]: Agreed, let's use the variable.

**Review by SergeGautherie — COMMENTED** [review]
I moved `DPRINT1()`s to the 2 called functions.

See also the added 'TODO' and 'FIXME' comments.

**Comment by SergeGautherie** [comment]
Debug output examples, before I filtered them out:
```
(/hal/halx86/legacy/bussupp.c:1571) HalGetBusDataByOffset(4, 0, 0xFF) failed (2 != 64)

(/hal/halx86/acpi/busemul.c:237) HalGetBusDataByOffset(4, 0, 0xFF) failed (2 != 64)
(/hal/halx86/acpi/busemul.c:246) HalGetBusDataByOffset(4, 1, 0x0) failed (0 != 64)
```

**Comment by SergeGautherie** [comment]
Was updated per comments. Now looking for new review.

**Pull request closed without merging**